### PR TITLE
only start servers after a manager became a leader

### PR DESCRIPTION
**Pull request by cmingxu**
Signed-off-by: kevin xu <cming.xu@gmail.com>

### CODE REVIEW

**Comment by upccup** [comment]
+1

File: src/manager/apiserver/server.go [hunk 1]
```diff
@@ -59,7 +59,11 @@ func (s *ApiServer) AppendRouter(routers ...router.Router) {
 }
 
 func (s *ApiServer) ListenAndServe() error {
-	var chError = make(chan error)
+	var wg sync.WaitGroup
+	var e error
+
+	wg.Add(2)
```
Comment (pwzgorilla) [inline src/manager/apiserver/server.go]: 这样做看不出任何必要 而且增加了复杂性

**Comment by pwzgorilla** [comment]
not good, see comments.

**Comment by cmingxu** [comment]
close this

**Pull request closed without merging**